File: python/hot100/30.两两交换链表中的节点.py
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy=ListNode(0,head)
        pre=dummy
        while pre.next and pre.next.next:
            temp1=pre.next
            temp2=temp1.next
            lat=temp2.next
            pre.next=temp2
            temp2.next=temp1
            temp1.next=lat
            pre=temp1
        return dummy.next

File: cpp/hot100/30.两两交换链表中的节点.cpp
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *dummy=new ListNode(0);
        dummy->next=head;
        ListNode *pre=dummy;
        while(pre->next&&pre->next->next){
            ListNode *temp1=pre->next->next;
            ListNode *temp2=pre->next;
            ListNode *lat=temp1->next;
            pre->next=temp1;
            temp1->next=temp2;
            temp2->next=lat;
            pre=temp2;
        }
        return dummy->next;
    }
};